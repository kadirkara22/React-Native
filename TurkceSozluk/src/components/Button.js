import { TouchableOpacity } from 'react-native'
import styled from 'styled-components'
import { compose, color, size, space, flexbox, layout, borderRadius,position } from 'styled-system'

const Button = styled(TouchableOpacity)(
    compose(
        space,
        size,
        color,
        flexbox,
        layout,
        borderRadius,
        position
    )
);


Button.defaultProps = {
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center"
}
export default Button