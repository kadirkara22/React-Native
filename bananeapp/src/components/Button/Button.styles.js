import { StyleSheet } from "react-native";
import colors from "../../styles/colors"
const base_style = StyleSheet.create({
    container: {
        padding: 10,
        margin: 10,
        borderRadius: 5,
        alignItems: "center"
    },
    title: {
        fontWeight: "bold",
        fontSize: 17
    }
})
export default {
    primary: StyleSheet.create({
        ...base_style,
        container: {
            ...base_style.container,
            backgroundColor: colors.darkgreen,
        },
        title: {
            ...base_style.title,
            color: "white",
        }
    }),
    secondary: StyleSheet.create({
        ...base_style,
        container: {
            ...base_style.container,
            backgroundColor: "white",
            borderWidth: 1,
            borderColor: colors.darkgreen,
        },
        title: {
            ...base_style.title,
            color: colors.darkgreen,

        }
    })
}

