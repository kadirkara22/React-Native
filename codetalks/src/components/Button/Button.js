import React from 'react'
import { View, Text, TouchableOpacity } from 'react-native'
import styles from "./Button.style"
const Button = ({ onPress, text, theme }) => {
    return (
        <TouchableOpacity style={styles[theme].container} onPress={onPress}>
            <Text style={styles[theme].title}>{text}</Text>
        </TouchableOpacity>
    )
}

export default Button
