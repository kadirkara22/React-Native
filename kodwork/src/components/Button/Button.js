import React from "react"
import { TouchableOpacity, Text, ActivityIndicator } from "react-native"
import styles from "./Button.styles"
const Button = ({ text, onPress }) => {
    return (
        <TouchableOpacity style={styles.container} onPress={onPress}>
            <Text style={styles.title}>{text}</Text>
        </TouchableOpacity>
    )
}

export default Button;
