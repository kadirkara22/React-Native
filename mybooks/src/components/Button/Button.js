import { isValid } from 'date-fns'
import React from 'react'
import { Text, TouchableOpacity } from 'react-native'
import styles from "./Button.style"
const Button = ({ onPress, theme, text, loading }) => {
    return (
        <TouchableOpacity onPress={onPress} style={styles[theme].container} disabled={loading} >
            <Text style={styles[theme].title}>{text}</Text>
        </TouchableOpacity >
    )
}

export default Button
