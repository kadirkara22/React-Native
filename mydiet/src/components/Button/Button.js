import React from 'react'
import { View, Text, TouchableOpacity } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialCommunityIcons'
import styles from "./Button.style"
const Button = ({ onPress, iconName, theme, iconTheme }) => {
    return (
        <TouchableOpacity onPress={onPress} style={styles[theme].container}>
            <Icon name={iconName} size={30} color="#33B24B" style={styles[iconTheme].icon} />
        </TouchableOpacity>
    )
}

export default Button
