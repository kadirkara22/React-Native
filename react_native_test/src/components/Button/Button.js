import React from 'react'
import { Text, TouchableOpacity } from 'react-native'
import styles from "./Button.style"
const Button = ({ title, onClick, theme = "primary" }) => {
    return (
        <TouchableOpacity testID="button-touchable" style={styles[theme].container} onPress={onClick}>
            <Text testID="button-title" style={styles[theme].title}>{title}</Text>
        </TouchableOpacity>
    )
}

export default Button
