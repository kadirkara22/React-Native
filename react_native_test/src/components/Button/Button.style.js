import { StyleSheet } from "react-native"

export default {
    primary: StyleSheet.create({
        container: {
            backgroundColor: "blue",
            margin: 10,
            padding: 10,
            borderRadius: 10,
            justifyContent: "center",
            alignItems: "center"

        },
        title: {
            fontWeight: "bold",
            color: "white",
        }
    }),
    secondary: StyleSheet.create({
        container: {
            borderWidth: 1,
            borderColor: "blue",
            margin: 10,
            padding: 10,
            borderRadius: 10,
            justifyContent: "center",
            alignItems: "center"

        },
        title: {
            fontWeight: "bold",
            color: "blue",
        }
    })
}