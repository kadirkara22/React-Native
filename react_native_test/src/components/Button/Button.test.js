import React from "react"
import { fireEvent, render } from "@testing-library/react-native"
import Button from "./Button"
import styles from "./Button.style"

test('should match with snapshot', () => {
    const comp = render(<Button />);
    expect(comp).toMatchSnapshot();
})

test('should render title correctly', () => {
    const testTitle = "test"
    const comp = render(<Button title={testTitle} />)

    const buttonText = comp.getByTestId("button-title").children[0];
    // console.log(buttonText)
    expect(buttonText).toBe(testTitle)
})

test('should trigerr onPress', () => {
    const mockFunction = jest.fn();
    const comp = render(<Button onClick={mockFunction} />)

    const buttonTouchable = comp.getByTestId('button-touchable');
    fireEvent(buttonTouchable, 'press')

    expect(mockFunction).toBeCalled()
    //expect(mockFunction).toBeCalledTimes(2) birden fazla tıklama olduğunda

})

test('should render default given theme style', () => {
    const comp = render(<Button />)

    const buttonStyle = comp.getByTestId('button-touchable').props.style
    expect(buttonStyle).toMatchSnapshot(styles.primary.container)
})

test('should render given theme style', () => {
    const selectedTheme = 'secondary'
    const comp = render(<Button theme={selectedTheme} />)

    const buttonStyle = comp.getByTestId('button-touchable').props.style
    expect(buttonStyle).toMatchSnapshot(styles[selectedTheme].container)
})
