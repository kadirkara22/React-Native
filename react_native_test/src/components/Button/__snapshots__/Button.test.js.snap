// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderRadius": 10,
      "justifyContent": "center",
      "margin": 10,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="button-touchable"
>
  <Text
    style={
      Object {
        "color": "white",
        "fontWeight": "bold",
      }
    }
    testID="button-title"
  />
</View>
`;

exports[`should render default given theme style 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "blue",
  "borderRadius": 10,
  "justifyContent": "center",
  "margin": 10,
  "opacity": 1,
  "padding": 10,
}
`;

exports[`should render given theme style 1`] = `
Object {
  "alignItems": "center",
  "borderColor": "blue",
  "borderRadius": 10,
  "borderWidth": 1,
  "justifyContent": "center",
  "margin": 10,
  "opacity": 1,
  "padding": 10,
}
`;
